src/msg_handler.c

```diff
@@ -0,0 +1,388 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#include "msg_handler.h"
+#include "global_vars.h"
+#include "output.h"
+#include "led.h"
+#include "fan.h"
+#include "buzzer.h"
+
+static char* itoa(int value, char* string);
+
+int parse_msg(char* buf, char* response, int max_len)
+{
+    char temp_str[11] = {0};  // for doing itoa conversions
+    response[0] = '\0';  // make a blank string
+
+    char* next_arg = strtok(buf, ":");
+    if (strcmp(next_arg, "OUT") == 0) {
+        next_arg = strtok(NULL, ":");
+        if (next_arg == NULL) {
+            strncat(response, "NACK:Missing output number", max_len);
+            return strlen(response);
+        }
+
+        int output_num;
+
+        if (isdigit((int)next_arg[0])) {
+            output_num = atoi(next_arg);
```

Comment:
What happens if I send something like `OUT:4294967296`?

Reply:
The output number is now bounded to ULONG_MAX and the bounds check below will throw it out as an invalid output number.